dask/array/core.py

```diff
@@ -5549,29 +5557,32 @@ class BlockView:
     An instance of ``da.array.Blockview``
     """
 
-    def __init__(self, array: Array) -> BlockView:
+    def __init__(self, array: Array) -> None:
         self._array = array
 
     def __getitem__(
         self, index: int | Sequence[int] | slice | Sequence[slice]
-    ) -> Array:
+    ) -> Array:  # type: ignore
         from dask.array.slicing import normalize_index
 
         if not isinstance(index, tuple):
-            index = (index,)
-        if sum(isinstance(ind, (np.ndarray, list)) for ind in index) > 1:
+            new_index = (index,)
+        else:
+            new_index = index  # type: ignore
+        if sum(isinstance(ind, (np.ndarray, list)) for ind in new_index) > 1:  # type: ignore
```

Comment:
Something's fishy here too. If the purpose was to accept ``np.ndarray | list``, why did the definition go for a much more relaxed Sequence?